execution_engine/src/core/engine_state/mod.rs

```diff
@@ -117,15 +119,8 @@ where
         &self.config
     }
 
-    pub fn get_protocol_data(
-        &self,
-        protocol_version: ProtocolVersion,
-    ) -> Result<Option<ProtocolData>, Error> {
-        match self.state.get_protocol_data(protocol_version) {
-            Ok(Some(protocol_data)) => Ok(Some(protocol_data)),
-            Err(error) => Err(Error::Exec(error.into())),
-            _ => Ok(None),
-        }
+    pub fn update_config(&mut self, new_config: EngineConfig) {
```

Comment:
I think we should put `EngineConfig` (or `WasmConfig` and `SystemConfig`) as part of `UpgradeConfig` - this way we'd enforce code change in node's contract runtime component - currently, the node wouldn't pass new `EngineConfig` as part of the upgrade.

I also think by doing that we should be able to remove `new_validator_slots` `new_auction_delay` `new_locked_funds_period_millis` `new_round_seigniorage_rate` `new_unbonding_delay`, and just take those values from appropriate configs

@Fraser999 @xcthulhu Should we still update the system contract's named keys for purposes of fast sync as part of upgrade? Would it be safe to wire up system contracts to access the EngineConfig properties directly?